### PR TITLE
A small amount of tidying-up factored out from PR #64648

### PR DESCRIPTION
**Pull request by alexreg**
As requested by @Mark-Simulacrum, I put this in a separate commit to make it easier to review. (As far as I can tell, no violations of the policy here, and they are simply in a separate PR because they're not directly related to the import of that PR.)

r? @Mark-Simulacrum

### CODE REVIEW

File: src/libsyntax_pos/lib.rs [hunk 1]
```diff
@@ -985,17 +985,16 @@ impl Encodable for SourceFile {
...
             let name: FileName = d.read_struct_field("name", 0, |d| Decodable::decode(d))?;
             let name_was_remapped: bool =
                 d.read_struct_field("name_was_remapped", 1, |d| Decodable::decode(d))?;
             let src_hash: u128 =
                 d.read_struct_field("src_hash", 2, |d| Decodable::decode(d))?;
             let start_pos: BytePos =
-                d.read_struct_field("start_pos", 4, |d| Decodable::decode(d))?;
-            let end_pos: BytePos = d.read_struct_field("end_pos", 5, |d| Decodable::decode(d))?;
-            let lines: Vec<BytePos> = d.read_struct_field("lines", 6, |d| {
+                d.read_struct_field("start_pos", 3, |d| Decodable::decode(d))?;
```
Comment (Mark-Simulacrum) [inline src/libsyntax_pos/lib.rs]: Could you track down where this discrepancy was first introduced (git blame or so) just to make sure we're not missing some intentional change here?

File: src/libsyntax_pos/lib.rs [hunk 2]
```diff
@@ -1024,18 +1023,18 @@ impl Decodable for SourceFile {
                 Ok(lines)
             })?;
             let multibyte_chars: Vec<MultiByteChar> =
-                d.read_struct_field("multibyte_chars", 7, |d| Decodable::decode(d))?;
```
Comment (Mark-Simulacrum) [inline src/libsyntax_pos/lib.rs]: And same with this change (git blame)

File: src/libsyntax_pos/lib.rs (hunk 1 above)
Comment (alexreg) [reply src/libsyntax_pos/lib.rs]: Sure. It was commit `0b4c2cccac30ebcd436e0cfd77b34019c40d4ce3`, from Dec 2017, which changed:

```
-            s.emit_struct_field("src_hash", 6, |s| self.src_hash.encode(s))?;
-            s.emit_struct_field("start_pos", 2, |s| self.start_pos.encode(s))?;
-            s.emit_struct_field("end_pos", 3, |s| self.end_pos.encode(s))?;
-            s.emit_struct_field("lines", 4, |s| {
+            s.emit_struct_field("src_hash", 2, |s| self.src_hash.encode(s))?;
+            s.emit_struct_field("stable_id", 3, |s| self.stable_id.encode(s))?;
+            s.emit_struct_field("start_pos", 4, |s| self.start_pos.encode(s))?;
+            s.emit_struct_field("end_pos", 5, |s| self.end_pos.encode(s))?;
+            s.emit_struct_field("lines", 6, |s| {
```

i.e. It simply removed the `stable_id` field, but forgot to change the sequence of indices. Looks innocuous to make this fix. :-)

File: src/libsyntax_pos/lib.rs (hunk 2 above)
Comment (alexreg) [reply src/libsyntax_pos/lib.rs]: Same commit as above. This is just a reordering of the indices to make it more readable/sane.

**Comment by alexreg** [comment]
Thanks!

**Pull request merged**